packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3233,8 +3233,8 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
 
   /// The configuration for this element.
   @override
-  Widget get widget => _widget!;
-  Widget? _widget;
+  Widget get widget => _widget;
```

Comment:
I thought we had documentation somewhere that was specifically telling people to override this with a getter that type casts to the right type. Obviously, that would need to be updated. But I can't find it, so maybe we don't?

Reply:
Ah! It's on RenderObjectElement: https://master-api.flutter.dev/flutter/widgets/RenderObjectElement-class.html

Can you update that doc there with the new practise?

Reply:
It may also be helpful to add a note to this doc string describing the pattern.

Reply:
I think we could probably make this API mostly private too. I would need to update the doc with a better pattern

Reply:
I've removed the public API

Reply:
https://master-api.flutter.dev/flutter/widgets/RenderObjectElement-class.html still needs to be updated, though. It's telling people to do the casting pattern for new RenderObjectElements that we are now discouraging.